client/db/src/bench.rs

```diff
@@ -156,7 +166,13 @@ impl<B: BlockT> StateBackend<HashFor<B>> for BenchmarkingState<B> {
 		child_info: ChildInfo,
 		key: &[u8],
 	) -> Result<Option<Vec<u8>>, Self::Error> {
-		self.state.borrow().as_ref().ok_or_else(state_err)?.child_storage(storage_key, child_info, key)
+		let state = self.state.borrow();
+		let db_state = state.as_ref().ok_or_else(state_err)?;
+		
+		let value = db_state.child_storage(storage_key, child_info, key)?;
+		let key = (storage_key.to_vec(), key.to_vec());
+		let value = self.state_usage_stats.tally_child_key_read(&key, value, false);
+		Ok(value)
 	}
 
 	fn exists_storage(&self, key: &[u8]) -> Result<bool, Self::Error> {
```

Comment:
This is the storage read too, is it not?

Reply:
Same for the child version

Reply:
Tbh I copied the work done in [#4562](https://github.com/paritytech/substrate/pull/4562/files).
I followed the same model as [storage_cache.rs](https://github.com/paritytech/substrate/blob/master/client/db/src/storage_cache.rs), where [storage](https://github.com/paritytech/substrate/blob/b14359cf5341a37028996a5002cf731a49312454/client/db/src/storage_cache.rs#L500) and [child_storage](https://github.com/paritytech/substrate/blob/b14359cf5341a37028996a5002cf731a49312454/client/db/src/storage_cache.rs#L543) update the usage stats, but not [exists_storage](https://github.com/paritytech/substrate/blob/b14359cf5341a37028996a5002cf731a49312454/client/db/src/storage_cache.rs#L576) nor [child_exists_storage](https://github.com/paritytech/substrate/blob/b14359cf5341a37028996a5002cf731a49312454/client/db/src/storage_cache.rs#L580) do.

Maybe I misunderstood and this is the case only because it's in CachingState impl? Or is it an issue in [storage_cache.rs](https://github.com/paritytech/substrate/blob/master/client/db/src/storage_cache.rs#L580) too?

Reply:
`exists_storage` calls `storage` there in real backend, in your code it does not, so @shawntabrizi is right